### DISCUSSION TITLE
OKRs Q2 2021

### DISCUSSION THREAD

**Discussion by mrmetaverse** [omigroup/omigroup#10] [General]
**EDIT: Date is set 05/05/21 6AM PDT and 6PM PDT:** 
Good morning. The OMI journey is both large in potential impact, and in contributors. We are looking to plan an OKR workshop to establish baseline Objectives and Key Results so that we may track our progress, as well as lay critical breadcrumbs for future contributors. 

**OKR's empower us with measurable goals**, however we are not expecting to achieve 100% of every OKR in a given quarter. OKRs encourage a flexible, and outcome driven mindset (rather than output driven). Our Objectives should tie directly to our overall vision, which I've taken a crack at crafting a first draft: _OMI's vision is to co-create an open set of standards for connecting the Metaverse. This includes it's users, stakeholders, and creators._ 

AngellXR is happy to facilitate this working session. We haven't set a date yet, and are still in planning. **Basically consider everything a draft, if you see an opportunity to improve something, please say something** 

We can drop OKR related readings and updates in this discussion thread. Cheers! 

Want to know more about OKRs? Check out this TED Talk from John Doerr: https://www.youtube.com/watch?v=L4N1q4RNi9I 
_ideas are easy, execution is everything_

**Comment by TrevorFSmith** [comment]
I support the idea of having concrete OKRs with a schedule.

Regarding the vision statement: At the risk of opening an unending semantic discussion, I haven't seen a solid consensus in the group about the meaning of "the Metaverse". Definitions range from "anything spatial on the Internet" to "teleporting myself and my inventory between first person immersive 3D platforms" and variations like "adding WebXR and payments to web pages".

**Reply by mrmetaverse** [reply]
Ah yes, good point. Some of those definitions are use cases and potential product specs. 

I appreciate that call out. Perhaps on of our objectives is to Define the Open Metaverse. With KRs like "discover and adopt a definition for The Metaverse, with a general consensus score of 90% or more" and "create a service diagram of 99% of the existing XR ecosystem," and "The definition is publically referenced and adopted by 5 organizations" 

Again in spit-balling and we haven't done the workshop together yet, but any thoughts?

**Reply by TrevorFSmith** [reply]
I suspect that we could relatively quickly agree on a subset of technologies (e.g. portable identity, bodies, and inventories) so instead of ratholing on "metaverse" in the mission statement we could leave it in the group name and in the mission be a bit more specific. I have mixed feelings about either path, though.

**Reply by mrmetaverse** [reply]
I am on the fence here too. Like do we dive straight into "we're making it possible to (optionally) federate the entire digital equivalent to the universe"? Obviously that's could use a little polish. 

This is another workshop worthy discussion. Maybe we find a regular weekly or bi-weekly workshop cadence to work through some of this. Vision and objectives are highly important ✨

**Reply by robertlong** [reply]
I have similar thoughts on this. I think we could agree upon a few core technologies/specs and focus the mission statement. Let's get this vision/mission/OKR workshop scheduled soon. We can discuss more in tomorrow's meeting.

**Reply by auscoi** [reply]
Maybe metaverse isn't the right term to head towards. What we're defining isn't so much another -verse as it is a layer or skin to an existing -verse. As I run it through my head I keep coming back to: are we simply updating the definition of the internet as we know it to include these newer techs and ideas or is the scope more about an application of standards and practices that will be put in place over top of our existing structures.

How large or narrow are we even trying to scope? Is limited to XR or only "direct participatory" experiences or is there some passive elements like would there be any 2D applications that will need to encompass "metaverse" standards. 

Just riffing :)

**Reply by mrmetaverse** [reply]
Great call outs, this is also a good reason for us to come to an agreed upon definition of metaverse, because I am of the school of thought that metaverse includes those 2D applications as well. Pretty much in my opinion, the metaverse is the digital counterpart of the universe, so apps, web, software, games, and XR would all fall in that realm for me. I also tend to conflate XR with the metaverse, in that any virtual, augmented, mixed, extended reality would fall under XR. 

I am also sure there could be dozens who disagree, and think this would be a really important discussion. 

I think this is something Trevor was calling attention to as well, so maybe this is one of our group OKRs or Tasks, Colloquially defining the metaverse.

**Comment by mrmetaverse** [comment]
Good resource: https://www.garyfox.co/okr-guide/

**Comment by humbletim** [comment]
Would it be possible (practical) to conduct initial OKR workshops as two parallel groups? Specifically thinking about time zone coverage -- like if we offered _both_ 6am and 6pm workshops so at least one time occurs reasonably close to local daytime hours anywhere.

**Reply by mrmetaverse** [reply]
thanks for the reminder. I am happy to conduct two sessions at 6AM PDT and 6PM PDT to accomdate for others. This workshop is flexible, and meant to be repeatable as we learn and grow. Two formats come to mind: 

Option I: 

6AM - OKR part I (1.5 hours) - Objectives, we will work as a group to aggregate key objectives, organize by theme, and discuss expected or potentially measurable outcomes. 

6PM - OKR part II (1.5 hours) - Key Results, we will resume from the efforts of the previous session, and this second group will assess the key themes, divide objectives into top-level, and specific-subset-level Objectives (the key here is top level, groups will have the option of setting sub-OKRs in alignment with these top level themes) Starting with the top level, we will discuss how we would know if we hit the given objective. What a lofty goal would be, and what a super "realistic" goal would be. We will repeat this process for the remaining items. 

We will present the OKRs on GH discussions for all to review and take a vote the following day. We will select an agreed upon number of OKR's, and teams can form around themes as needed. 

------ or ------ 
Option II:

We take 2 days, and do a 6AM (1 hour) Objective I and a 6PM (1 hour) Objective I session, followed by a 6AM (1 hour) Key Results II session and a 6PM (1 hour) Key Results II session. 

I am happy to work around what we believe will best help the group. My suggestion would be option I: 2 sessions total, one for O and one for KR and then an async presentation of the options, Q&A and adoption of our first round.

We should then reassess our progress every 2-4 weeks, eliminate OKRs if we find any are out of scope, and redraft our OKRs every 3-4 months.

**Reply by auscoi** [reply]
I support splitting the timezones to include the other hemisphere.

Also I'd vote for option #1 as herding cats twice is twice as hard as herding cats once.

**Comment by mrmetaverse** [comment]
Ok based on some conversation, the 2 session option might be best rather than the 4 session option. 

I am available the following dates at **6AM PDT **and** 6PM PDT**: 
Friday 4/30/21 
Sunday 5/2/21
Monday 5/3/21 
Tuesday 5/4/21 
*Wednesday 5/5/21 (just added 4/27)

6AM - OKR part I (1.5 hours) - **Objectives**, we will work as a group to aggregate key objectives, organize by theme, and discuss expected or potentially measurable outcomes.

6PM - OKR part II (1.5 hours) - **Key Results**, we will resume from the efforts of the previous session, and this second group will assess the key themes, divide objectives into top-level, and specific-subset-level Objectives (the key here is top level, groups will have the option of setting sub-OKRs in alignment with these top level themes) Starting with the top level, we will discuss how we would know if we hit the given objective. What a lofty goal would be, and what a super "realistic" goal would be. We will repeat this process for the remaining items.

We will present the OKRs on GH discussions for all to review and take a vote after 24-48 hours. We will select an agreed upon number of OKR's, and teams can form around themes as needed. Teams will self assemble and establish Theme-specifc OKR's or epics that fit with our top-level OKRs.

**Reply by robertlong** [reply]
Personally I would like next Tuesday. But I can do Friday, Monday, or Tuesday at those times. Our survey showed that people were least available on Tuesdays though.

Love the idea of capturing the results of these conversations and continuing the conversation async on GH discussions. That sounds great :+1:

**Reply by robertlong** [reply]
Here's the results of that final survey again:
![image](https://user-images.githubusercontent.com/1753624/116153825-c8dd7e00-a69c-11eb-85f6-1d1b3a42e20a.png)

There's one with more time slots as well but it's a bit hard to read. The final survey contained all the most popular times.

**Reply by robertlong** [reply]
Also, this workshop should replace our next meeting. We really need to meet about OKRs/goal setting before we continue our weekly meeting. Can we get this scheduled by Tomorrow afternoon so we have enough time to get the word out?

**Reply by mrmetaverse** [reply]
I could do next Wednesday, and Wednesdays look like pretty good days given this sample.

**Comment by mrmetaverse** [comment]
Want to know more about OKRs? Check out this TED Talk from John Doerr: https://www.youtube.com/watch?v=L4N1q4RNi9I
ideas are easy, execution is everything

**Comment by mrmetaverse** [comment]
That was a productive first session, thank you to those who were able to attend. We elected not to record the OKR sessions to promote psychological safety, and open discussion. The artifact can be viewed in figjam here: https://www.figma.com/file/VZZG4K3OigHdXxuWxKnuqU/OMi-OKR-Workshop-I?node-id=0%3A1

This evening (for me at least) we will establish high level themes, and how we will measure our success as a group. And then... its time to make magic happen!

**Comment by mrmetaverse** [comment]
### Happy Friday, Metaverse. 

We will continue the OKR workshop asynchronously until our next gathering. We made a lot of progress in our two sessions together. You can check out the Figjam space [here](https://www.figma.com/file/HcSUQRYUHjTcVyGIA0fu1u/OMi-OKR-Workshop-Part-II?node-id=0%3A1). [Figjam ](https://www.figma.com/figjam/)is a new collaborative whiteboard product from Figma.

The next step is to review our Objective statements. We made an effort to capture overarching themes from the thematic clusters we created in our workshop.  Please review and discuss, or if you have a suggestion for a great way to discuss each of these 5 themes in an organized way, please reach out! The 5 objective statements we have are as follows:

**1. We collaborate to make the metaverse more human.**

**2. We empower the people making the metaverse.** 

**3. We are creating a cooperative, sustainable culture of innovation.** 

**4. We incrementally deliver value, with intent.** 

**5. We must achieve resiliency through diversity, and freedom of choice.**

![OKR-omiObjectiveDrafts](https://user-images.githubusercontent.com/63426722/117510531-3a011900-af5a-11eb-8d0d-f17581e0165a.png)

Keep in mind, we are looking for _high level_, and _measurable_ objectives. Once we agree on a core set of objectives, we will begin describing how we know if we have achieved (or how we are progressing towards) an objective. Specific tactics, or "_hows_," and best practices most likely do not apply at this level. Those important concepts will apply at the execution level of these strategies and the tactics that are formed to achieve these objectives. 

We are also not looking for perfection. OKRs are not meant to be achieved 100%. If we score **.6** on a scale of  **0.0 - 1.0**, that is pretty good! We will revisit and refine our objectives every 3-4 months, and artifacts like our formal vision statement, and roadmap, will be informed by these high level goals we set as a fledgling organization. 

Please explore, and give these concepts great consideration. Is there something we are missing, or that can be refined? Do these make you feel inspired? 

Thank you for your commitment to this adventure together. 

For more information on OKR's check out "Measure What Matters: OKRs: The Simple Idea that Drives 10x Growth,"
 by John Doerr, or this article here: [link]( https://www.whatmatters.com/faqs/okr-meaning-definition-example/#:~:text=The%20definition%20of%20%E2%80%9COKRs%E2%80%9D%20is,encourage%20engagement%20around%20measurable%20goals 
)

**Reply by robertlong** [reply]
We'll be figuring out how to measure our progress towards these objectives in our next meeting. However, we can also start that discussion here.

1. We collaborate to make the metaverse more human.
  In our discussions we were focused on human-centered design. Designing for people-first and ensuring that our work respects people's privacy and is accessible to all. I think we need to gather some research in this field to clarify what we mean here. We should have specific pieces of research that we can reference throughout our work. In a similar fashion to what we did in [creating a list of standards/protocols](https://github.com/omigroup/OMI/discussions/12) that members of the community can reference, we can create a list of research/researchers. We also should be reaching out to more researchers in this field and inviting them to join our organization. One metric I'd like to propose for tracking this is to have at least 3 researchers on privacy, ethical, and inclusive design as active members of OMI.
  
2. We empower the people making the metaverse.
  The core of this objective was ensuring a great experience for creators/implementers learning about OMI and creating content for it. A lot of this will be centered around documentation and ensuring there are good resources for people to integrate our work into their projects. At this point we don't have that work, but we can start gathering requirements for what resources OMI could provide to ease the transition to creating interoperable metaverse content. I think there's a ton of discussion we could start having around content interoperability today. For example, can we write up documentation/create tooling to make it easier to make cross platform avatars/scenes? glTF for example has really wide adoption and VRM is an avatar format based on glTF that is gaining wider adoption. Could we feature metaverse platforms that support these formats and help content creators publish interoperable assets?

3. We are creating a cooperative, sustainable culture of innovation.
  I started some GitHub conversations for this objective. We talked about incentives, how do we get people to participate in OMI long term? We also talked about having bi-weekly show and tell meetups and monthly lightning talk meetups. We also need to talk about how we want to structure OMI outside of protocols/standards work. I created a GitHub discussion for that [here](https://github.com/omigroup/OMI/discussions/31).

4. We incrementally deliver value, with intent.
  This was really about not biting off more than we can chew. The metaverse is a big, lofty, open-ended concept. We need to focus on pieces of it rather than all of it at once. To start I think we need to have a more serious discussion about our scope of work. What are a few concepts that we have in common across metaverse platforms? Identity, inventory, friends lists, and content portability are some that have popped up in the Discord. Within each of those, what is a minimal specification that would have a big impact on interoperability. I created a [github discussion for determining this scope of work](https://github.com/omigroup/OMI/discussions/28) and we'll discuss this at the next meeting as well.
  I'd also challenge us to work with existing protocols as much as possible. How can we utilize and extend existing efforts versus starting new ones from scratch.

5. We must achieve resiliency through diversity, and freedom of choice.
  This objective was about a number of things. We discussed coming up with modular protocols, so that different platforms could agree to implement parts of the OMI specs, but not all of them. Every platform is going to be different and that's something we should encourage. Not every platform with support XR, not every platform will be 3D, there can be multiple views into the metaverse. We also talked about federation. Federation to me is decentralization through choice. By designing the right protocols, you can have the choice of who you trust with your data. Platform owners have the choice to centralize parts of their platform and decentralize others. Technology will evolve, but we can still design long-lasting protocols defining the data flowing through these systems.

**Reply by CyberGuardian-XRSI** [reply]
Hi All,

I would like to petition that OMI looks into adopting the goals set by XRSI's Diversity and Inclusion effort, The Cyber XR Coalition. We plan to turn this into a pledge for the industry so everyone can subscribe to it for their apps, platforms, artwork, and metaverses. These standards built by the Cyber XR coalition have been used by several stakeholders including the Govt of Australia to create their positioning statement for XR for the whole country (AU). Here are the goals:

1. Leave no one behind.
2. Be accessible:
    Everyone must be able to participate in the digital society.
3. Protect identities:
    Users must be able to participate in the digital society no matter their gender, ethnicity, birthplace, or cultural or political 
    beliefs, ensuring discrimination and biases are mitigated and not further reinforced.
4. Keep everyone safe and secure:
    Shaping rules and practices that enable a secure and resilient immersive environment.
5. Build new rules to promote trust:
    Developing new, flexible, participatory governance mechanisms to complement traditional policy and regulation 
    in a domain that’s in a state of constant evolution.

Ref: https://cyberxr.org/research-and-standards/
https://xrsi.org/publication/the-xrsi-privacy-framework
https://xrsi.org/publication/extend-reality-with-awareness-xr-safety-and-privacy-guide-for-artists

P.S XRSI remains open to any suggestions and edits before we launch them as a pledge.

I look forward to the community feedback - Founder -XRSI -Kavya Pearlman

**Reply by mrmetaverse** [reply]
Great idea Kavya, I'm sure we could learn a lot from your efforts. Just took a look with humbletim. These are mission-worthy objectives and principles and I'd like to continue the discussion. Would you be willing to discuss how we might adopt these practices in OMi for 10-15 min at an upcoming OMi weekly meeting? 

I know people receive and process information in different ways so maybe an open discussion could lead to a dedicated github discussions or Discord thread to resume discussion afterwards. 

I also see a lot of these principles playing a role in team-wide or OMi-wide working agreements. Maybe we can talk about ways of incorporating this in our work together.

**Comment by mrmetaverse** [comment]
Update: OKR's are now assembled in a single doc, and I am working on language now. Getting the information organized in such a way was the majority of the work. But now it's all ready for plugging in many of the themes we identified as a group. However, some of the key results are still missing. We have a good number of tasks, stories, and epic-fodder were crafted as a bi-product of our efforts. Now it's about making it all actionable and suited to our needs. 

I will keep working on this tonight, and hope we can review as a group for about 15-20 of this weeks session. We are so close! 

https://www.figma.com/file/VdJyo53j68SmGZbuApiE9Y/OMI-OKRs?node-id=2%3A25

**Reply by mrmetaverse** [reply]
![OMIOKRs](https://user-images.githubusercontent.com/63426722/118419839-315fbf80-b68b-11eb-8891-a9ca5c5be7ac.png)

**Comment by mrmetaverse** [comment]
![image](https://user-images.githubusercontent.com/63426722/119376459-f4e32380-bc89-11eb-8d23-fa623ec73ccb.png)
